game_management/game_input_handler.py

```diff
@@ -0,0 +1,560 @@
+import random
+import typing
+
+import pyxel
+
+from board import Board
+from calculator import clamp, complete_construction, attack_setl
+from catalogue import get_available_improvements, get_available_blessings, get_available_unit_plans, PROJECTS
+from game_management.game_controller import GameController
+from game_management.game_state import GameState
+from menu import MainMenuOption, SetupOption, WikiOption
+from models import Construction, OngoingBlessing, CompletedConstruction, Heathen, GameConfig, OverlayType, Faction, \
+    ConstructionMenu, Project
+from movemaker import set_player_construction
+from overlay import SettlementAttackType, PauseOption
+from saving.game_save_manager import save_game, load_game, get_saves
+
+
+def on_key_down(game_controller: GameController, game_state: GameState, is_ctrl_key: bool):
```

Comment:
I think this and the up version of it are probably named a bit confusingly - on_key_down and on_key_up could refer to any key being pressed or released.

Reply:
Good call, addressed

Reply:
LGTM!